android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -82,6 +82,7 @@ public void onNetworkItemSelected(NetworkSelectorItem networkSelectorItem) {
                                 Log.e(TAG, "Could not set network");
                             }
                         });
+                finish();        
```

Comment:
`finish` needs to be moved inside a response of `jsonRpcService.setNetwork`. As that is an async request and it could be that the object is destroyed before the call is actually done.